internal/helm/dependency_manager_test.go

```diff
@@ -38,180 +43,593 @@ var (
+		wantChartFunc              func(g *WithT, c *helmchart.Chart)
+		wantErr                    string
+	}{
+		//{
+		//	// TODO(hidde): add various happy paths
+		//},
+		//{
+		//	// TODO(hidde): test Chart.lock
+		//},
+		{
+			name:    "build failure returns error",
+			baseDir: "testdata/charts",
+			path:    "helmchartwithdeps",
+			wantErr: "failed to add remote dependency 'grafana': no chart repository for URL",
+		},
+		{
+			name:    "no dependencies returns zero",
+			baseDir: "testdata/charts",
+			path:    "helmchart",
+			want:    0,
+		},
 	}
-	if err := dm.Build(context.TODO()); err != nil {
-		t.Errorf("Build() should return nil")
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			chart, err := loader.Load(filepath.Join(tt.baseDir, tt.path))
+			g.Expect(err).ToNot(HaveOccurred())
+
+			got, err := NewDependencyManager(chart, tt.baseDir, tt.path).
+				WithRepositories(tt.repositories).
+				WithChartRepositoryCallback(tt.getChartRepositoryCallback).
+				Build(context.TODO())
+
+			if tt.wantErr != "" {
+				g.Expect(err).To(HaveOccurred())
+				g.Expect(err.Error()).To(ContainSubstring(tt.wantErr))
+				g.Expect(got).To(BeZero())
+				return
+			}
+
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(got).To(Equal(tt.want))
+			if tt.wantChartFunc != nil {
+				tt.wantChartFunc(g, chart)
+			}
+		})
 	}
 }
 
-func TestBuild_WithLocalChart(t *testing.T) {
+func TestDependencyManager_build(t *testing.T) {
 	tests := []struct {
 		name    string
-		dep     helmchart.Dependency
-		wantErr bool
-		errMsg  string
+		deps    map[string]*helmchart.Dependency
+		wantErr string
 	}{
 		{
-			name: "valid path",
-			dep: helmchart.Dependency{
+			name: "error remote dependency",
+			deps: map[string]*helmchart.Dependency{
+				"example": {Repository: "https://example.com"},
+			},
+			wantErr: "failed to add remote dependency",
+		},
+		{
+			name: "error local dependency",
+			deps: map[string]*helmchart.Dependency{
+				"example": {Repository: "file:///invalid"},
+			},
+			wantErr: "failed to add remote dependency",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			dm := &DependencyManager{
+				baseDir: "testdata/charts",
+			}
+			err := dm.build(context.TODO(), tt.deps)
+			if tt.wantErr != "" {
+				g.Expect(err).To(HaveOccurred())
+				return
+			}
+
+			g.Expect(err).ToNot(HaveOccurred())
+		})
+	}
+}
+
+func TestDependencyManager_addLocalDependency(t *testing.T) {
+	tests := []struct {
+		name     string
+		dep      *helmchart.Dependency
+		wantErr  string
+		wantFunc func(g *WithT, c *helmchart.Chart)
+	}{
+		{
+			name: "local dependency",
+			dep: &helmchart.Dependency{
 				Name:       chartName,
 				Version:    chartVersion,
-				Repository: chartLocalRepository,
+				Repository: "file://../helmchart",
+			},
+			wantFunc: func(g *WithT, c *helmchart.Chart) {
+				g.Expect(c.Dependencies()).To(HaveLen(1))
```

Comment:
`wantFunc` can be removed from here and the above struct, since it's not used in the test below.